src/include/sof/audio/component.h

```diff
@@ -699,8 +699,32 @@ static inline struct comp_dev *comp_alloc(const struct comp_driver *drv,
 #define comp_get_drvdata(c) \
 	(c->priv_data)
 
-#if defined UNIT_TEST || defined __ZEPHYR__
+#if defined UNIT_TEST || defined __ZEPHYR__  || CONFIG_LIBRARY_STATIC
 #define DECLARE_MODULE(init)
+
+/* declared modules */
+void sys_comp_asrc_init(void);
```

Comment:
random thought, any reason we are not putting a section in the memory map to dynamic registration in a library?

Reply:
xt-testbench simulator usage is static, but see ```lrg/topic/plugin``` branch as that uses dynamic loading via ALSA plugin.